azure/apis/storage/v1alpha1/types.go

```diff
@@ -272,16 +300,65 @@ func (c *Container) GetContainerName() string {
 	return util.ConditionalStringFormat(c.Spec.NameFormat, string(c.GetUID()))
 }
 
-// ParseContainerSpec from properties map key/values
-func ParseContainerSpec(p map[string]string) *ContainerSpec {
-	return &ContainerSpec{
-		ReclaimPolicy:    v1alpha1.ReclaimRetain,
-		Metadata:         util.ParseMap(p["metadata"]),
-		NameFormat:       p["nameFormat"],
-		PublicAccessType: parsePublicAccessType(p["publicAccessType"]),
-	}
+// +kubebuilder:object:root=true
+
+// ContainerList - list of the container objects
+type ContainerList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Container `json:"items"`
 }
 
+// ContainerClassSpecTemplate is the Schema for the resource class
+type ContainerClassSpecTemplate struct {
+	v1alpha1.ResourceClassSpecTemplate `json:",inline"`
+	ContainerParameters                `json:",inline"`
+}
+
+var _ resource.Class = &ContainerClass{}
+
+// +kubebuilder:object:root=true
+
+// ContainerClass is the Schema for the resource class
+// +kubebuilder:printcolumn:name="PROVIDER-REF",type="string",JSONPath=".specTemplate.providerRef.name"
+// +kubebuilder:printcolumn:name="RECLAIM-POLICY",type="string",JSONPath=".specTemplate.reclaimPolicy"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+type ContainerClass struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	SpecTemplate ContainerClassSpecTemplate `json:"specTemplate,omitempty"`
+}
+
+// GetReclaimPolicy of this ContainerClass.
+func (i *ContainerClass) GetReclaimPolicy() v1alpha1.ReclaimPolicy {
+	return i.SpecTemplate.ReclaimPolicy
+}
+
+// SetReclaimPolicy of this ContainerClass.
+func (i *ContainerClass) SetReclaimPolicy(p v1alpha1.ReclaimPolicy) {
+	i.SpecTemplate.ReclaimPolicy = p
+}
+
+// +kubebuilder:object:root=true
+
+// ContainerClassList contains a list of cloud memorystore resource classes.
+type ContainerClassList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ContainerClass `json:"items"`
+}
+
+// ParseContainerSpec from properties map key/values
```

Comment:
can this function be removed instead of just commented out?

Reply:
Absolutely! Just missed that one